Autopilot/PathManager/Inc/waypointManager.hpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * Waypoint Manager Methods and Helpers
+ * Author: Dhruv Rawat
+ * Created: November 2020
+ * Last Updaetd: November 2020
+ */
+
+#ifndef WAYPOINT_MANAGER_H
+#define WAYPOINT_MANAGER_H
+
+#include <stdlib.h>
+
+#define ORBIT_FOLLOWING 0
+#define LINE_FOLLOWING 1
+
+#define PI 3.14159265359
+#define MAX_PATH_APPROACH_ANGLE PI/2
+
+#define PATH_BUFFER_SIZE 100
+
+//Waypoint Types
+#define DEFAULT_WAYPOINT 0 // Used to navigate flight path
+#define WAYPOINT_UNUSED 1
+#define HOLD_WAYPOINT 2 // Circle
+
+// Placeholder: 
+typedef struct {
+    /*
+        Contains:
+            long 
+            lat
+            alt
+            heading
+    */    
+} _Gps_Data;
+
+// Stores error codes for the waypoint manager
+enum _WaypointStatus {WAYPOINT_SUCCESS = 0, WAYPOINT_NOT_FOUND, WAYPOINT_PARAMETERS_NOT_DEFIND, UNDEFINED_FALIURE};
+
+// Used in the waypointBufferStatus array to signal which elements are free
+enum _WaypointBufferStatus {FREE = 0, FULL};
+
+//Used to specify the modification type when updating the waypointBuffer array
+enum _WaypointBufferUpdateType {ADD_WAYPOINT = 0, UPDATE_WAYPOINT, INSERT_WAYPOINT, DELETE_WAYPOINT};
+
+
+/*
+* Structure stores information about the waypoints along our path to the destination and back. 
+*/
+typedef struct {
```

Comment:
A lot of my comments on the members of this struct aren't valid if this isn't the case, but this is the struct the state machine would use to ad a new waypoint, right ?

Reply:
yeah, so if the user/state machine will need to Create the waypoints using this structure and then pass them into the methods of the waypoint manager. 

We should move this struct to a separate file, but I am putting it here so my code acc compiles lol